src/main/java/com/sender/email/MyConfig.java

```diff
@@ -0,0 +1,29 @@
+package com.sender.email;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+@Configuration
+public class MyConfig {
+
+    @Bean
+    public TaskScheduler getScheduler(){
+        TaskScheduler scheduler = new ThreadPoolTaskScheduler();
```

Comment:
Make scheduler variable of type `ThreadPoolTaskScheduler`. Thus you can get rid of casts and still keep it general because of method return type